src/main/java/com/example/decafe/HelloController.java

```diff
@@ -0,0 +1,577 @@
+    public ImageView upgradeCoffeeImageView;
+    public ImageView upgradeCakeImageView;
+    public ImageView upgradePlayerImageView;
+
+    // Labels used for collision detection management
+    public Label table1;
+    public Label table2;
+    public Label table3;
+    public Label table4;
+    public Label plantsAbove;
+    public Label countRight;
+    public Label countBelow;
+    public Label customerTop1;
+    public Label customerTop2;
+    public Label customerTop3;
+    public Label customerTop4;
+    public Label customerBot1;
+    public Label customerBot2;
+    public Label customerBot3;
+    public Label plant;
+    public Label edgeBot;
+    public Label edgeTop;
+    public Label edgeLeft;
+    public Label edgeRight;
+
+    //for the customers
+    //smiley images
+    public ImageView smileyFirst;
+    public ImageView smileySecond;
+    public ImageView smileyThird;
+    public ImageView smileyFourth;
+    public ImageView smileyFifth;
+    public ImageView smileySixth;
+    public ImageView smileySeventh;
+
+    //coin images
+    public ImageView coinFirst;
+    public ImageView coinSecond;
+    public ImageView coinThird;
+    public ImageView coinFourth;
+    public ImageView coinFifth;
+    public ImageView coinSixth;
+    public ImageView coinSeventh;
+
+    //order labels
+    public ImageView orderlabel1 = new ImageView();
+    public ImageView orderlabel2 = new ImageView();
+    public ImageView orderlabel3 = new ImageView();
+    public ImageView orderlabel4 = new ImageView();
+    public ImageView orderlabel5 = new ImageView();
+    public ImageView orderlabel6 = new ImageView();
+    public ImageView orderlabel7 = new ImageView();
+
+    //customer images
+    public ImageView first;
+    public ImageView second;
+    public ImageView third;
+    public ImageView fourth;
+    public ImageView fifth;
+    public ImageView sixth;
+    public ImageView seventh;
+
+    // for end screen
+    public ImageView gameStartButton;
+    public ImageView cofiBrewImage;
+    public ImageView playAgainImage;
+    public ImageView backToStartImage;
+    public Label labelCredits;
+    public ImageView endScreenBackground;
+    public ImageView quitEndScreenImage;
+
+    // Player object created to change Images and movement Speed
+    public Player CofiBrew = new Player("CofiBrewUp.png", "CofiBrewCakeLeft.png", "CofiBrewCoffeeLeft.png", 4);
+    // Game object used to control main methods
+    public Game Play;
+    // Label array used for collision detection management
+    private Label[] collisions;
+    // Timer used to spawn customers or make them leave
+    public Timer controllerTimer = new Timer();
+
+    public File f = new File("");
+    public String musicFile = f.getAbsolutePath() + File.separator + "src" + File.separator + "main" + File.separator + "resources" + File.separator + "com" + File.separator + "example" + File.separator + "decafe" + File.separator + "backgroundmusic.mp3";
```

Comment:
Define a constant instead of duplicating the literals "resources", "example" and "decafe" 7 times (they are also on line 335, 404, 416, 427, 499, 523)